### PR TITLE
chore: guard regression with trtllm_bf16_routed_moe tests

### PR DESCRIPTION
**Pull request by rosenrodt**
## 📌 Description

**Update: bug no longer reproducible on ToT since https://github.com/flashinfer-ai/flashinfer/pull/2803 fixed it. The only useful part left is the regression testing.**

Old title was `fix: segfault using packed topk_id/weight as trtllm_bf16_routed_moe input in DeepSeek routing`

Old description:

When deciding whether to enter the main routing kernel inside the MoE, switch the condition from `mPtrTopKIds == nullptr` to `mPtrScores != nullptr`.

`mPtrTopKIds == nullptr` is true for the "packed" pre-routed inputs so it will go down the main routing kernel path. It is the wrong code path.  As `mPtrScores` is the routing logits, the condition should be `mPtrScores != nullptr`.

## 🔍 Related Issues

Found when integrating FlashInfer BF16 TRTLLM MoE in https://github.com/NVIDIA/TensorRT-LLM/pull/12557

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

## Summary by CodeRabbit

* **Tests**
  * Added explicit DeepSeekV3 routing coverage for MoE equivalence checks.
  * Enhanced routed-MoE test harness with configurable routing parameters and deterministic seeding to improve validation and reproducibility.

### CODE REVIEW

**Comment by rosenrodt** [comment]
Bug no longer reproducible on ToT since #2803 fixed it. The only useful part left is the regression testing.

**Comment by rosenrodt** [comment]
cc @aleozlx

**Comment by aleozlx** [comment]
circling back to this. would you pls help resolve merge conflicts?

**Review by aleozlx — APPROVED** [review]
